### ISSUE TITLE
mesheryctl makefile is not able to detect fedora/Linux of binary

### ISSUE THREAD

**Issue by anandrkskd** [layer5io/meshery#1026]
**Description**
Currently makefile for mesheryctl is not able to detect Linux as user os and is creating binary for darwin.

**Expected Behaviour**
makefile should auto detect the OS and create binary accordingly.

**Screenshots**
![Screenshot from 2020-05-19 10-26-53](https://user-images.githubusercontent.com/34443651/82332424-00f72e00-9a03-11ea-892c-5be75df03881.png)

**Environment:**
 - OS: Fedora Linux
 - Device: laptop

**To Reproduce**
1. Go to mesheryctl dir
2. run `make`
3. run the binary inside './bin/mesheryctl'
**Additional Context**
Related to [PR](https://github.com/layer5io/meshery/pull/1021)

**Comment by leecalcote** [comment]
This issue is related to this PR - https://github.com/meshery/meshery/pull/1003. @Aisuko may want to have a look here.

**Comment by Aisuko** [comment]
Thanks for report the issue. Actually, I'd like to let the developer make a decision on which platform as a target to build. But, It's ok to add auto-detection. I will deal with this.

**Comment by Aisuko** [comment]
@leecalcote I modified the multiple builds scripts and I saw the build scripts add by PR #1021 (So crazy with lots of files changed huh ). And I saw it work well for most situations. 

https://github.com/meshery/meshery/blob/b5f37c653c9d791fe1f7e30aee1cb2dc474ff48b/mesheryctl/Makefile#L1-L4

So, the issue above may not happen again. And the  `auto-detect the OS` is a new feature request(I didn't find the feature label, so I add another reference label).
